MIGRATING.md

```diff
@@ -0,0 +1,71 @@
+# Okta Java Management SDK Migration Guide
+ 
+This SDK uses semantic versioning and follows Okta's [library version policy](https://developer.okta.com/code/library-versions/). In short, we don't make breaking changes unless the major version changes!
+
+## Migrating from 1.x.x to 2.0.0
+
+Version 2.0.0 of this SDK introduces a number of breaking changes from previous versions. 
+In addition to many new classes/interfaces, some existing class/interfaces are no longer backward compatible due to method renaming and signature changes, see [Breaking Changes](#breaking-changes).
+
+### Breaking Changes
+
+Following is a list of changes that break backward compatibility in version 2.0.0.
+
+- `com.okta.sdk.client.Client`
+   - Renamed `createRule` to `createGroupRule`
+   - Renamed `getRule` to `getGroupRule`
+   - Renamed `listRules` to `listGroupRules`
+   
+- `com.okta.sdk.resource.log.LogEventList` 
+   - `getLogs(String until, String since, String filter, String q, String sortOrder)`
+   Signature changed to `getLogs(Date since, Date until, String filter, String q, String sortOrder)`
+   
+- `com.okta.sdk.resource.group.GroupList` 
+   - `listGroups(String q, String filter, String expand)`
+   Signature changed to `listGroups(String q, String filter)`
+
+- `com.okta.sdk.resource.application.SwaThreeFieldApplicationSettingsApplication`
+   - `getTargetUrl` Renamed to `getTargetURL`
+   - `setTargetUrl` Renamed to `setTargetURL`
+
+- `com.okta.sdk.resource.group.Group`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.group.GroupType`
+   
+- `com.okta.sdk.resource.group.rule.GroupRule`
+   - `delete(Boolean removeUsers)` Signature changed to `delete()`
+   - Removed `getAllGroupsValid` and `setAllGroupsValid` (property `allGroupsValid` is removed)
+   - Removed `getEmbedded` (property `_embedded` is removed)
```

Comment:
does this remove functionality or was the previous spec incorrect?

Reply:
This is a removed property as part of this spec revision. Not sure why it is removed.

Reply:
@bretterer do you have an idea why this property __embedded_ is removed from the new spec?

Reply:
`_embedded` is not documented for this model

Reply:
@bdemers @bretterer what is your suggestion here? should we tell the reader that the property is now removed or?

Reply:
worse case, just say it isn't documented (that seems really bad, but 🤷 )

Reply:
updated